src/coreComponents/physicsSolvers/NonlinearSolverParameters.hpp

```diff
@@ -172,14 +175,23 @@ class NonlinearSolverParameters : public dataRepository::Group
   /// Factor by which the time step will be cut if a timestep cut is required.
   real64 m_timeStepCutFactor;
 
-  /// number of times that the time-step had to be cut
+  /// Number of times that the time-step had to be cut
   integer m_numdtAttempts;
 
   /// number of times that the configuration had to be changed
   integer m_numConfigurationAttempts;
 
   /// Max number of times that the configuration can be changed
   integer m_maxNumConfigurationAttempts;
+
+  // Simulation statistics
+
+  /// Total number of successful Newton iterations
+  integer m_totalSuccessfulNewtonNumIterations;
+
+  /// Total number of wasted Newton iterations
+  integer m_totalWastedNewtonNumIterations;
+
```

Comment:
This should (at least) go in its own `class`.
Here we're mixing input/configuration parameters and meta-results.
As a side effect you could wrap a `LvArray< integer >`, one index for each information, so you should be able to use it in `TimeHistory` too (I know it's a workaround 😉 )

Reply:
The PR no longer deals with the solver parameters, I thought it would be easier to do something separately.